### ISSUE TITLE
v1.12.0~rc.3-1 generates non-existent device mapping

### ISSUE THREAD

**Issue by Clockwork-Muse** [NVIDIA/nvidia-container-toolkit#40]
The output of `nvidia-ctk cdi generate` in v1.12.0~rc.3-1 includes a device mapping that doesn't exist on my host machine; `/dev/dri/controlD64`.  Commenting out the device appears to make it work (I'm able to get blender 3.4 to render headless)...

Host: Ubuntu 22.04
Host Driver: nvidia-driver-525
Toolkit: v1.12.0~rc.3-1
Container runtime: Podman 4.3.1 (installed via the kubic repository)
Container image: docker.io/nvidia/cudagl:11.4.2-runtime-ubuntu20.04

Output of `nvidia-ctk cdi generate --format yaml`:
```sh
INFO[0000] Selecting /var/run/nvidia-persistenced/socket as /var/run/nvidia-persistenced/socket 
WARN[0000] Could not locate /var/run/nvidia-fabricmanager/socket: pattern /var/run/nvidia-fabricmanager/socket not found 
WARN[0000] Could not locate /tmp/nvidia-mps: pattern /tmp/nvidia-mps not found 
INFO[0000] Using driver version 525.60.11               
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvoptix.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-vulkan-producer.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-tls.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-rtcore.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-ptxjitcompiler.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-opticalflow.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-opencl.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-nvvm.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-ngx.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-ml.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-glvkspirv.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-glsi.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-glcore.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-fbc.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-encode.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-eglcore.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-compiler.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-cfg.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvidia-allocator.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libnvcuvid.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libcudadebugger.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libcuda.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libGLX_nvidia.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libGLESv2_nvidia.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libGLESv1_CM_nvidia.so.525.60.11 
INFO[0000] found 64-bit driver lib: /usr/lib/x86_64-linux-gnu/libEGL_nvidia.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-tls.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-ptxjitcompiler.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-opticalflow.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-opencl.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-nvvm.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-ml.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-glvkspirv.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-glsi.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-glcore.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-fbc.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-encode.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-eglcore.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvidia-compiler.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libnvcuvid.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libcuda.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libGLX_nvidia.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libGLESv2_nvidia.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libGLESv1_CM_nvidia.so.525.60.11 
INFO[0000] found 32-bit driver lib: /usr/lib/i386-linux-gnu/libEGL_nvidia.so.525.60.11 
INFO[0000] Selecting /dev/nvidia-modeset as /dev/nvidia-modeset 
INFO[0000] Selecting /dev/nvidia-uvm-tools as /dev/nvidia-uvm-tools 
INFO[0000] Selecting /dev/nvidia-uvm as /dev/nvidia-uvm 
INFO[0000] Selecting /dev/nvidiactl as /dev/nvidiactl   
INFO[0000] Selecting /usr/lib/x86_64-linux-gnu/libnvidia-egl-gbm.so.1.1.0 as /usr/lib/x86_64-linux-gnu/libnvidia-egl-gbm.so.1.1.0 
INFO[0000] Selecting /usr/share/glvnd/egl_vendor.d/10_nvidia.json as /usr/share/glvnd/egl_vendor.d/10_nvidia.json 
INFO[0000] Selecting /usr/share/vulkan/icd.d/nvidia_icd.json as /usr/share/vulkan/icd.d/nvidia_icd.json 
INFO[0000] Selecting /usr/share/vulkan/implicit_layer.d/nvidia_layers.json as /usr/share/vulkan/implicit_layer.d/nvidia_layers.json 
INFO[0000] Selecting /usr/share/egl/egl_external_platform.d/15_nvidia_gbm.json as /usr/share/egl/egl_external_platform.d/15_nvidia_gbm.json 
INFO[0000] Selecting /usr/share/egl/egl_external_platform.d/10_nvidia_wayland.json as /usr/share/egl/egl_external_platform.d/10_nvidia_wayland.json 
WARN[0000] Could not locate /lib/firmware/nvidia/525.60.11/gsp.bin: pattern /lib/firmware/nvidia/525.60.11/gsp.bin not found 
INFO[0000] Selecting /usr/bin/nvidia-smi as /usr/bin/nvidia-smi 
INFO[0000] Selecting /bin/nvidia-smi as /bin/nvidia-smi 
INFO[0000] Selecting /usr/bin/nvidia-debugdump as /usr/bin/nvidia-debugdump 
INFO[0000] Selecting /bin/nvidia-debugdump as /bin/nvidia-debugdump 
INFO[0000] Selecting /usr/bin/nvidia-persistenced as /usr/bin/nvidia-persistenced 
INFO[0000] Selecting /bin/nvidia-persistenced as /bin/nvidia-persistenced 
INFO[0000] Selecting /usr/bin/nvidia-cuda-mps-control as /usr/bin/nvidia-cuda-mps-control 
INFO[0000] Selecting /bin/nvidia-cuda-mps-control as /bin/nvidia-cuda-mps-control 
INFO[0000] Selecting /usr/bin/nvidia-cuda-mps-server as /usr/bin/nvidia-cuda-mps-server 
INFO[0000] Selecting /bin/nvidia-cuda-mps-server as /bin/nvidia-cuda-mps-server 
---
cdiVersion: 0.4.0
containerEdits:
  deviceNodes:
  - path: /dev/nvidia-modeset
  - path: /dev/nvidia-uvm-tools
  - path: /dev/nvidia-uvm
  - path: /dev/nvidiactl
  hooks:
  - args:
    - nvidia-ctk
    - hook
    - update-ldcache
    - --folder
    - /usr/lib/x86_64-linux-gnu
    - --folder
    - /usr/lib/i386-linux-gnu
    hookName: createContainer
    path: /usr/bin/nvidia-ctk
  - args:
    - nvidia-ctk
    - hook
    - chmod
    - --mode
    - "755"
    - --path
    - /dev/dri
    hookName: createContainer
    path: /usr/bin/nvidia-ctk
  mounts:
  - containerPath: /var/run/nvidia-persistenced/socket
    hostPath: /var/run/nvidia-persistenced/socket
    options:
    - ro
    - nosuid
    - nodev
    - bind
    - noexec
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-egl-gbm.so.1.1.0
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-egl-gbm.so.1.1.0
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/share/egl/egl_external_platform.d/15_nvidia_gbm.json
    hostPath: /usr/share/egl/egl_external_platform.d/15_nvidia_gbm.json
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/share/egl/egl_external_platform.d/10_nvidia_wayland.json
    hostPath: /usr/share/egl/egl_external_platform.d/10_nvidia_wayland.json
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/share/glvnd/egl_vendor.d/10_nvidia.json
    hostPath: /usr/share/glvnd/egl_vendor.d/10_nvidia.json
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/share/vulkan/icd.d/nvidia_icd.json
    hostPath: /usr/share/vulkan/icd.d/nvidia_icd.json
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/share/vulkan/implicit_layer.d/nvidia_layers.json
    hostPath: /usr/share/vulkan/implicit_layer.d/nvidia_layers.json
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvoptix.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvoptix.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-vulkan-producer.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-vulkan-producer.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-tls.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-tls.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-rtcore.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-rtcore.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-ptxjitcompiler.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-ptxjitcompiler.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-opticalflow.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-opticalflow.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-opencl.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-opencl.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-nvvm.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-nvvm.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-ngx.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-ngx.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-ml.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-ml.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-glvkspirv.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-glvkspirv.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-glsi.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-glsi.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-glcore.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-glcore.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-fbc.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-fbc.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-encode.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-encode.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-eglcore.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-eglcore.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-compiler.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-compiler.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-cfg.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-cfg.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvidia-allocator.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvidia-allocator.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libnvcuvid.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libnvcuvid.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libcudadebugger.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libcudadebugger.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libcuda.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libcuda.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libGLX_nvidia.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libGLX_nvidia.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libGLESv2_nvidia.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libGLESv2_nvidia.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libGLESv1_CM_nvidia.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libGLESv1_CM_nvidia.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/x86_64-linux-gnu/libEGL_nvidia.so.525.60.11
    hostPath: /usr/lib/x86_64-linux-gnu/libEGL_nvidia.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-tls.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-tls.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-ptxjitcompiler.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-ptxjitcompiler.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-opticalflow.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-opticalflow.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-opencl.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-opencl.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-nvvm.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-nvvm.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-ml.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-ml.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-glvkspirv.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-glvkspirv.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-glsi.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-glsi.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-glcore.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-glcore.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-fbc.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-fbc.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-encode.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-encode.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-eglcore.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-eglcore.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvidia-compiler.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvidia-compiler.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libnvcuvid.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libnvcuvid.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libcuda.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libcuda.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libGLX_nvidia.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libGLX_nvidia.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libGLESv2_nvidia.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libGLESv2_nvidia.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libGLESv1_CM_nvidia.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libGLESv1_CM_nvidia.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/lib/i386-linux-gnu/libEGL_nvidia.so.525.60.11
    hostPath: /usr/lib/i386-linux-gnu/libEGL_nvidia.so.525.60.11
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/bin/nvidia-smi
    hostPath: /usr/bin/nvidia-smi
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /bin/nvidia-smi
    hostPath: /bin/nvidia-smi
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /bin/nvidia-debugdump
    hostPath: /bin/nvidia-debugdump
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/bin/nvidia-persistenced
    hostPath: /usr/bin/nvidia-persistenced
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /bin/nvidia-cuda-mps-control
    hostPath: /bin/nvidia-cuda-mps-control
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /bin/nvidia-cuda-mps-server
    hostPath: /bin/nvidia-cuda-mps-server
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/bin/nvidia-debugdump
    hostPath: /usr/bin/nvidia-debugdump
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /bin/nvidia-persistenced
    hostPath: /bin/nvidia-persistenced
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/bin/nvidia-cuda-mps-control
    hostPath: /usr/bin/nvidia-cuda-mps-control
    options:
    - ro
    - nosuid
    - nodev
    - bind
  - containerPath: /usr/bin/nvidia-cuda-mps-server
    hostPath: /usr/bin/nvidia-cuda-mps-server
    options:
    - ro
    - nosuid
    - nodev
    - bind
devices:
- containerEdits:
    deviceNodes:
    - path: /dev/nvidia0
    - path: /dev/dri/card0
    - path: /dev/dri/controlD64
    - path: /dev/dri/renderD128
    hooks:
    - args:
      - nvidia-ctk
      - hook
      - create-symlinks
      - --link
      - ../card0::/dev/dri/by-path/pci-0000:10:00.0-card
      - --link
      - ../renderD128::/dev/dri/by-path/pci-0000:10:00.0-render
      hookName: createContainer
      path: nvidia-ctk
  name: gpu0
- containerEdits:
    deviceNodes:
    - path: /dev/nvidia0
    - path: /dev/dri/card0
    - path: /dev/dri/controlD64
    - path: /dev/dri/renderD128
    hooks:
    - args:
      - nvidia-ctk
      - hook
      - create-symlinks
      - --link
      - ../card0::/dev/dri/by-path/pci-0000:10:00.0-card
      - --link
      - ../renderD128::/dev/dri/by-path/pci-0000:10:00.0-render
      hookName: createContainer
      path: nvidia-ctk
  name: all
kind: nvidia.com/gpu
```

**Comment by elezar** [comment]
@Clockwork-Muse thanks for the bugreport. I'm glad that this helps you to enable headless rendering in Blender.

I have a merge request (https://gitlab.com/nvidia/container-toolkit/container-toolkit/-/merge_requests/260) that should address the but that you're seeing by only injecting device nodes that actually exist. If you would be able to confirm with an nvidia-ctk executable built from the MR, that would be very helpful.

Any other comments would also be appreciated.

**Comment by Clockwork-Muse** [comment]
Using the packages built from CI (specifically `nvidia-container-toolkit-base`) gives a working config.

**Comment by elezar** [comment]
@Clockwork-Muse we have released our `v1.12.0` packages. Please reopen this issue if you're still having problems.
